apparelo/apparelo/doctype/grn/grn.py

```diff
@@ -12,11 +12,15 @@
 
 class GRN(Document):
 	def validate(self):
-		self.po=None
 		self.get_po()
 	def on_submit(self):
 		pr=self.create_purchase_receipt()
 		msgprint(_("{0} created").format(comma_and("""<a href="#Form/Purchase Receipt/{0}">{1}</a>""".format(pr.name, pr.name))))
+	
+	def on_cancel(self):
+		self.db_set("docstatus",2)
+		msgprint(_("{0} cancelled").format(comma_and("""<a href="#Form/GRN/{0}">{1}</a>""".format(self.name, self.name))))
+	
```

Comment:
similarly, we need to cancel the PR that was created with this GRN...

Reply:
Here too.